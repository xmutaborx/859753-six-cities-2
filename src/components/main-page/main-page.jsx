import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import ActionCreator from '../../store/action-creator';
import {getCityOffers, getCitiesList} from '../../store/selectors';

import Header from '../header/header.jsx';
import CitiesList from '../cities-list/cities-list.jsx';
import Sorting from '../sorting/sorting.jsx';
import OffersList from '../offer-list/offers-list.jsx';
import CitiesMap from '../cities-map/cities-map.jsx';
import MainEmpty from '../main-empty/main-empty.jsx';

const MainPage = (props) => {
  const {
    city,
    availableOffers,
    availableCities,
    changeCity,
    offers} = props;

  if (!offers.length) {
    return <MainEmpty />;
  }

  return (
    <div className="page page--gray page--main">
      <Header />
      <main className="page__main page__main--index">
        <CitiesList
          cities={availableCities}
          currentCity={city}
          onChangeCity={changeCity}
        />
        <div className="cities">
          <div className="cities__places-container container">
            <section className="cities__places places">
              <h2 className="visually-hidden">Places</h2>
              <b className="places__found">{availableOffers.length} places to stay in {city}</b>
              <Sorting />
              <OffersList offers={availableOffers} />
            </section>
            <div className="cities__right-section">
              <section className="cities__map map">
                <CitiesMap offersList={availableOffers} />
              </section>
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  city: state.city,
  offers: state.offers,
  availableCities: getCitiesList(state),
  availableOffers: getCityOffers(state),
});

const mapDispatchToProps = (dispatch) => ({
  changeCity: (city) => dispatch(ActionCreator.changeCity(city)),
});

MainPage.propTypes = {
  city: PropTypes.string,
  availableOffers: PropTypes.array,
  availableCities: PropTypes.array,
  changeCity: PropTypes.func,
  offers: PropTypes.array
};

export {MainPage};
export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
