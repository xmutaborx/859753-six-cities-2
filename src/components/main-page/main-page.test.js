import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from "enzyme-to-json";

import {MainPage} from './main-page.jsx';
import OFFER_MOCK from '../../mocks/offer-mock';

Enzyme.configure({adapter: new Adapter()});

it(`Main Page correctly renders`, () => {
  const tree = shallow(<MainPage
    city={`Moscow`}
    availableOffers={OFFER_MOCK}
    availableCities={[`Moscow`, `Berlin`]}
    changeCity={() => {}}
    offers={OFFER_MOCK}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
