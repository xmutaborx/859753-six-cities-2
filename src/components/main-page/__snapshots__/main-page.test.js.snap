// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page correctly renders 1`] = `
<div
  className="page page--gray page--main"
>
  <Connect(Header) />
  <main
    className="page__main page__main--index"
  >
    <CitiesList
      cities={
        Array [
          "Moscow",
          "Berlin",
        ]
      }
      currentCity="Moscow"
      onChangeCity={[Function]}
    />
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2
             places to stay in 
            Moscow
          </b>
          <Connect(WithSorting) />
          <Connect(WithActiveItem)
            offers={
              Array [
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.370216,
                      "longitude": 4.895168,
                      "zoom": 10,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "goods": Array [
                    "Heating",
                    "Kitchen",
                    "Cable TV",
                    "Washing machine",
                    "Coffee machine",
                    "Dishwasher",
                  ],
                  "host": Object {
                    "avatar_url": "img/1.png",
                    "id": 3,
                    "is_pro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "img/1.png",
                    "img/2.png",
                  ],
                  "is_favorite": false,
                  "is_premium": false,
                  "location": Object {
                    "latitude": 52.35514938496378,
                    "longitude": 4.673877537499948,
                    "zoom": 8,
                  },
                  "max_adults": 4,
                  "preview_image": "img/1.png",
                  "price": 120,
                  "rating": 4.8,
                  "title": "Beautiful & luxurious studio at great location",
                  "type": "apartment",
                },
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.370216,
                      "longitude": 4.895168,
                      "zoom": 10,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "goods": Array [
                    "Heating",
                    "Kitchen",
                    "Cable TV",
                    "Washing machine",
                    "Coffee machine",
                    "Dishwasher",
                  ],
                  "host": Object {
                    "avatar_url": "img/1.png",
                    "id": 3,
                    "is_pro": true,
                    "name": "Angelina",
                  },
                  "id": 2,
                  "images": Array [
                    "img/1.png",
                    "img/2.png",
                  ],
                  "is_favorite": false,
                  "is_premium": false,
                  "location": Object {
                    "latitude": 52.35514938496378,
                    "longitude": 4.673877537499948,
                    "zoom": 8,
                  },
                  "max_adults": 2,
                  "preview_image": "img/1.png",
                  "price": 400,
                  "rating": 5,
                  "title": "Beautiful & luxurious studio at great location",
                  "type": "apartment",
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Connect(CitiesMap)
              offersList={
                Array [
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatar_url": "img/1.png",
                      "id": 3,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "img/1.png",
                      "img/2.png",
                    ],
                    "is_favorite": false,
                    "is_premium": false,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "max_adults": 4,
                    "preview_image": "img/1.png",
                    "price": 120,
                    "rating": 4.8,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatar_url": "img/1.png",
                      "id": 3,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 2,
                    "images": Array [
                      "img/1.png",
                      "img/2.png",
                    ],
                    "is_favorite": false,
                    "is_premium": false,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "max_adults": 2,
                    "preview_image": "img/1.png",
                    "price": 400,
                    "rating": 5,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
